### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-09-15

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#21]
# iPAS AI 應用規劃師科目3練習 - 2026-09-15（第 11 輪）

本次共 **10 題**。請直接點選每個選項前的核取方塊作答，**不需留言、不需進入編輯畫面**。

- 單選題：只勾選 **1 個**你認為最正確的選項。
- 多選題：依題目標示的數量勾選（例如「可勾選 3 項」）。
- 作答方式：直接點擊選項前的 `- [ ]` 核取方塊即可勾選（變成 `- [x]`）。
- 批改方式：**全部勾選完成後，請將本 issue 的 label 改為 `ai-analysis`**，系統即會進行批改與錯題分析。

本回側重「L233 建模與參數調校」，並涵蓋「L234 機器學習治理」與跨領域考點。情境題分別以「物流與交通」及「公部門與智慧城市」為背景。

## 第 1 題〔單選〕L23301 數據準備與特徵工程

某交通違規偵測資料集嚴重不平衡（違規僅約 2%）。團隊打算使用 SMOTE 過採樣來平衡類別，以下哪一種做法最正確、且不會造成資料洩漏（Data Leakage）？

- [ ] (A) 在 Train/Test Split 之前，對整份資料集（含測試集）做 SMOTE，確保測試集也平衡
- [ ] (B) 對測試集也一併做過採樣，讓評估指標數字更好看
- [ ] (C) 先做 Train/Test Split，僅對訓練集做 SMOTE 過採樣，測試集維持原始分布
- [ ] (D) 直接刪除多數類別的大部分樣本，且測試集同步刪除以維持平衡

---

## 第 2 題〔單選〕L23302 模型訓練與調校

在眾多特徵中，若希望模型訓練後能自動做特徵選擇，使部分不重要特徵的權重被壓縮為 **0**（得到稀疏模型），最適合採用下列哪一種正則化方法？

- [ ] (A) L2 正則化（Ridge）
- [ ] (B) L1 正則化（Lasso）
- [ ] (C) Dropout
- [ ] (D) Early Stopping

---

## 第 3 題〔單選 · 程式判讀〕L23303 模型評估指標

閱讀下列程式碼，判斷輸出結果（1 = 異常，0 = 正常）：

```python
from sklearn.metrics import precision_score, recall_score

y_true = [1, 1, 1, 1, 1, 0, 0, 0, 0, 0]
y_pred = [1, 1, 1, 0, 0, 1, 0, 0, 0, 0]

print(round(precision_score(y_true, y_pred), 2),
      round(recall_score(y_true, y_pred), 2))
```

- [ ] (A) 0.75 0.6
- [ ] (B) 0.6 0.75
- [ ] (C) 0.75 0.75
- [ ] (D) 0.6 0.6

---

## 第 4 題〔情境分析 · 單選〕L23303 模型評估指標（物流與交通）

某貨運公司建立模型，預測「哪些包裹會在轉運過程中遺失或嚴重破損」。實際上會出問題的包裹極少（約 1%），但只要漏掉一件未預警的高價值包裹，賠償與商譽成本都很高。團隊發現模型的 Accuracy 高達 99%，正討論是否可以上線。下列判斷與做法何者最正確？

- [ ] (A) Accuracy 已達 99%，代表模型很好，可直接上線
- [ ] (B) 應只看 Precision，確保每一次預警都準確無誤
- [ ] (C) 改用非監督分群取代分類任務，即可解決不平衡問題
- [ ] (D) 資料極度不平衡會使 Accuracy 失真（全判為正常也有 99%）；應優先關注 Recall 以降低漏報，並輔以混淆矩陣、Precision、F1 綜合評估

---

## 第 5 題〔多選（可勾選 3 項）〕L23301 數據準備與特徵工程

下列哪些演算法在訓練前通常**需要先做特徵縮放**（標準化／正規化）？（可勾選 3 項）

- [ ] (A) K-Means 分群
- [ ] (B) 決策樹（Decision Tree）
- [ ] (C) KNN（K 近鄰）
- [ ] (D) 隨機森林（Random Forest）
- [ ] (E) 以梯度下降訓練的神經網路

---

## 第 6 題〔單選〕L23401 資料與模型治理

關於機器學習專案中的「資料血緣（Data Lineage）與版本控管」，下列敘述何者最正確？

- [ ] (A) 記錄資料來源、前處理流程、特徵版本與模型版本，有助於問題追溯、稽核與結果重現，是負責任治理的一環
- [ ] (B) 只要模型準確率夠高，就不需要記錄訓練資料的來源與版本
- [ ] (C) 資料血緣只有在模型出錯之後才需要，平時無須維護
- [ ] (D) 血緣紀錄應包含所有原始個資明文，方便日後隨時查閱比對

---

## 第 7 題〔多選（可勾選 3 項）〕L23402 AI 風險與倫理

為降低模型對特定族群的偏見、提升公平性，下列哪些做法**適當**？（可勾選 3 項）

- [ ] (A) 檢查訓練資料是否存在歷史偏差（Historical Bias）或抽樣偏差（Selection Bias）
- [ ] (B) 以公平性指標分群評估模型表現（例如比較各族群的核准率、假陽性率）
- [ ] (C) 直接刪除弱勢族群的測試資料，讓整體 Accuracy 看起來更好
- [ ] (D) 建立人為複核與申訴機制，於高風險決策保留 human-in-the-loop
- [ ] (E) 僅以整體 Accuracy 作為唯一驗收指標

---

## 第 8 題〔情境分析 · 單選〕L23401 資料與模型治理（公部門與智慧城市）

某市政府智慧城市平台計畫串接市民的用水、用電與門禁刷卡等資料，訓練模型預測「獨居長者是否出現異常狀況（如跌倒、失聯）」，以便主動關懷。專案啟動階段，最應**優先**確認的事項是下列何者？

- [ ] (A) 盡量蒐集越多個資越好，資料量大模型才會準
- [ ] (B) 先把資料公開給所有市府單位共用，加速跨局處協作
- [ ] (C) 先確認資料蒐集與使用的合法性、告知同意，僅取所需的最小資料，並落實去識別化與存取權限控管
- [ ] (D) 先全力追求最高準確率，隱私與合規問題等模型上線後再處理

---

## 第 9 題〔單選〕L23201 機器學習原理

關於偏差（Bias）與變異（Variance）的權衡，下列敘述何者正確？

- [ ] (A) 高偏差通常對應過擬合，代表模型過度記憶訓練資料
- [ ] (B) 高變異通常對應欠擬合，代表模型太過簡單
- [ ] (C) 只要不斷增加模型複雜度，就一定能同時降低偏差與變異
- [ ] (D) 高偏差對應欠擬合（模型太簡單）、高變異對應過擬合（對訓練資料太敏感），需在兩者之間取得權衡

---

## 第 10 題〔單選 · 程式判讀〕L23203 深度學習原理與框架

閱讀下列 PyTorch 程式碼，判斷輸出的張量形狀：

```python
import torch
import torch.nn as nn

model = nn.Sequential(
    nn.Linear(6, 12),
    nn.ReLU(),
    nn.Linear(12, 4),
)

x = torch.randn(32, 6)
print(model(x).shape)
```

- [ ] (A) torch.Size([32, 12])
- [ ] (B) torch.Size([32, 4])
- [ ] (C) torch.Size([6, 4])
- [ ] (D) torch.Size([32, 6])
